### PR TITLE
Add support for initContainers in job and task manager.

### PR DESCRIPTION
**Pull request by chrismoos**
This change adds support for `initContainers` in `jobManager` and `taskManager`. This allows users to initialize the job and task managers if necessary prior to starting.

Can address issue in #234.

### CODE REVIEW

File: controllers/flinkcluster_converter.go [hunk 1]
```diff
@@ -753,6 +755,52 @@ func convertFromSavepoint(jobSpec *v1beta1.JobSpec, clusterStatus *v1beta1.Flink
 	return jobSpec.FromSavepoint
 }
 
+func convertJobManagerInitContainers(jobManagerSpec *v1beta1.JobManagerSpec) []corev1.Container {
+	var initContainers = []corev1.Container{}
+	// Add jobManagerSpec level volume mounts to each init container if there is no
+	// conflict.
+	for _, initContainer := range jobManagerSpec.InitContainers {
```
Comment (functicons) [inline controllers/flinkcluster_converter.go:762]: Can we extract this logic into a helper method which can be shared by all 3 methods?

File: controllers/flinkcluster_converter.go [hunk 2]
```diff
@@ -753,6 +755,52 @@ func convertFromSavepoint(jobSpec *v1beta1.JobSpec, clusterStatus *v1beta1.Flink
...
+		for _, jobMount := range jobManagerSpec.VolumeMounts {
+			var conflit = false
```
Comment (functicons) [inline controllers/flinkcluster_converter.go:764]: s/conflit/conflict

**Comment by functicons** [comment]
The test is failing, could you run `make test` and fix it?

File: controllers/flinkcluster_converter.go (hunk 1 above)
Comment (chrismoos) [reply controllers/flinkcluster_converter.go]: This is done.

File: controllers/flinkcluster_converter.go (hunk 2 above)
Comment (chrismoos) [reply controllers/flinkcluster_converter.go]: Fixed.

**Comment by chrismoos** [comment]
@functicons Updated PR with changes, tests should be passing now as well.

**Pull request merged**